sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -280,13 +280,32 @@ class JacksonParser(
         case VALUE_STRING =>
           UTF8String.fromString(parser.getText)
 
-        case _ =>
+        case other =>
           // Note that it always tries to convert the data as string without the case of failure.
-          val writer = new ByteArrayOutputStream()
-          Utils.tryWithResource(factory.createGenerator(writer, JsonEncoding.UTF8)) {
-            generator => generator.copyCurrentStructure(parser)
+          val startLocation = parser.getTokenLocation
+          startLocation.contentReference().getRawContent match {
```

Comment:
Is there an existing API to get the remaining content as string? Also, would it work with multi-line JSON?

Reply:
I was not able to find such an existing API -- there is `JacksonParser.getText` but that appears to simply get the current value if it's a string value.

Reply:
wrt. multiline JSON, I have added a test to cover this